aws/services/rds/resource.ftl

```diff
@@ -324,4 +384,46 @@
             }
         )
     /]
+
+    [#list links as linkId, link ]
+        [#if link?is_hash]
+            [#local linkTarget = getLinkTarget(occurrence, link, false) ]
+            [@debug message="Link Target" context=linkTarget enabled=false /]
+
+            [#if !linkTarget?has_content]
+                [#continue]
+            [/#if]
+
+            [#local linkTargetCore = linkTarget.Core ]
+            [#local linkTargetConfiguration = linkTarget.Configuration ]
+            [#local linkTargetResources = linkTarget.State.Resources ]
+            [#local linkTargetAttributes = linkTarget.State.Attributes ]
+            [#local linkTargetSolution = linkTargetConfiguration.Solution]
+
+            [#if (link.Role+" ")?lower_case?starts_with("publish ") ]
+                [#local linkRole = (link.Role+" ")?lower_case?keep_after("publish ")?trim ]
+                [#if linkRole?has_content]
+                    [#local linkRoles = (linkRole?split(","))![] ]
+                [#else]
+                    [#local linkRoles = [] ]
+                [/#if]
+                [@cfResource
+                    id=formatId(id, "cluster", linkId)
+                    type="AWS::RDS::EventSubscription"
+                    properties=
+                        {
+                            "Enabled" : true,
+                            "SnsTopicArn" : linkTargetAttributes["ARN"],
+                            "SourceIds" : [ getReference(id) ],
+                            "SourceType" : "db-cluster"
+                        } +
+                        attributeIfContent(
+                            "EventCategories",
+                            linkRoles
+                        )
+                    outputs=RDS_EVENT_OUTPUT_MAPPINGS
+                /]
+            [/#if]
+        [/#if]
+    [/#list]
```

Comment:
This shouldn't be part of a resource and should actually be in the component instead. The resource macros are for creating a particular resource and potentially be indpendent from the occurrence processing. 

This is also a lot duplication between the cluster and the instance, when it could be handled in the setup macro.

Reply:
Are there any permissions requirements? Usually there needs to be an inbound role on a component so the SNS topic's resource policy can be udpated.

Reply:
That will mean that most macros in resource will just be a call to cfResource won't it? I don't understand, but done

Permissions don't appear to be required from my testing